\documentclass[fleqn,10pt]{wlscirep}
\usepackage[T1]{fontenc}
\usepackage{lineno}
\usepackage{setspace}
\usepackage[utf8]{inputenc}


\usepackage{polyglossia}
\setdefaultlanguage[variant=british]{english}

%\usepackage[Latin, Phonetics, Diacritics, SpacingModifierLetters]{ucharclasses}

\usepackage{fontspec}
\defaultfontfeatures{Scale=MatchLowercase,Mapping=tex-text}


\usepackage{tipa}
%\newfontfeature{IPA}{+mgrk}
%\setmainfont[IPA]{DejaVu Sans}
%\newfontfamily\dejavuserif[IPA]{\DejaVu Serif}

%\setTransitionTo{IPAExtensions}{\dejavuserif}
%\setTransitionFrom{IPAExtentions}{\normalfont}
%\setTransitionTo{CombiningDiacriticalMarks}{\dejavuserif}
%\setTransitionFrom{CombiningDiacriticalMarks}{\normalfont}
%\setTransitionTo{SpacingModifierLetters}{\dejavuserif}
%\setTransitionFrom{SpacingModifierLetters}{\normalfont}

\usepackage{amsmath}

\usepackage{newfloat}
\DeclareFloatingEnvironment[name={Supplementary Figure S},fileext=lof]{supfig}
%\usepackage[euler]{textgreek}

\usepackage{hyperref}


\newcommand{\musqueam}{x\super{w}m\textschwa{}\texttheta{}k\super{w}\textschwa{}\'{y}\textschwa{}m}

\title{Sticky Pi, an AI-powered smart insect trap for community chronoecology}

\author[1,2,3,*]{Quentin Geissmann}
\author[4]{Paul K. Abram}
\author[3]{Di Wu}
\author[1,2]{Cara H. Haney}
\author[3]{Juli Carrillo}
\affil[1]{Department of Microbiology and Immunology, The University of British Columbia, Vancouver, Canada V6T 1Z3}
\affil[2]{Michael Smith Laboratories, The University of British Columbia, Vancouver, Canada V6T 1Z4}
\affil[3]{The University of British Columbia, Faculty of Land and Food Systems, Centre for Sustainable Food Systems and Biodiversity Research Centre, Vancouver, British Columbia, Unceded \musqueam Musqueam Territory, Canada, V6T 1Z3}
%$\overset{,}{y}$
\affil[4]{Agriculture and Agri-Food Canada, Agassiz Research and Development Centre, 6947 Highway \#7, Agassiz, BC, V0M 1A0, Canada}

\affil[*]{\href{mailto:qgeissmann@gmail.com}{qgeissmann@gmail.com}}



\keywords{Biosurveillance, Deep learning, \emph{Drosophila suzukii}, Circadian rhythm}


\begin{abstract}

%\begin{linenumbers}
Circadian clocks are paramount to insect survival and drive many aspects of their physiology and behaviour. 
While insect circadian behaviours have been extensively studied in the laboratory, their circadian activity within natural settings is poorly understood. 
The study of circadian activity necessitates measuring biological variables (\emph{e.g.}, locomotion) at high frequency (\emph{i.e.}, at least several times per hour) over multiple days, which has mostly confined insect chronobiology to the laboratory. 
In order to study insect circadian biology in the field, we developed the Sticky Pi, a novel, autonomous, open-source, insect trap that acquires images of sticky cards every twenty minutes.
Using custom deep-learning algorithms, we automatically and accurately scored \emph{where, when and which} insects were captured.
First, we validated our device in controlled laboratory conditions with a classic chronobiological model organism, \emph{Drosophila melanogaster}.
Then, we deployed an array of Sticky Pis to the field to characterise the daily activity of an agricultural pest, \emph{Drosophila suzukii}, and its parasitoid wasps. Finally, we demonstrate the wide scope of our smart trap by describing the sympatric arrangement of insect temporal niches in a community, without targeting particular taxa \emph{a priori}.
Together, the automatic identification and high sampling rate of our tool provide biologists with unique data that impacts research far beyond chronobiology; with applications to biodiversity monitoring and pest control as well as fundamental implications for phenology, behavioural ecology, and ecophysiology.
We released the Sticky Pi project as an open community resource on \href{https://doc.sticky-pi.com}{https://doc.sticky-pi.com}.
%\end{linenumbers}
	
\end{abstract}

\pagebreak
\begin{document}
\linenumbers
\setstretch{1.8}

\flushbottom
\maketitle

%
\thispagestyle{empty}



\section*{Introduction}

In order to fully characterise ecological communities, we must go beyond mere species inventories and integrate functional aspects such as interspecific interactions and organisms’ behaviours through space and time\cite{bro-jorgensen_linking_2019, cordero-rivera_behavioral_2017}. % 
Chronobiology, the study of biological rhythms, has shown that circadian (\emph{i.e.}, internal) clocks play ubiquitous and pivotal physiological roles, and that the daily timing of most behaviours matters enormously\cite{patke_molecular_2020}. Therefore, understanding not only which species are present, but also \emph{when} they are active adds a crucial, functional, layer to community ecology.

The emerging field of “chronoecology”, has begun to integrate chronobiological and ecological questions to reveal important phenomena \cite{halle_chronoecology_2000,helm_two_2017}. For instance, certain prey can respond to predators by altering their diel activity \cite{van_der_veen_flexible_2017}, parasites may manipulate their host’s clock to increase their transmission \cite{westwood_evolutionary_2019}, foraging behaviours are guided by the circadian clock \cite{jain_time-restricted_2018}, and, over evolutionary timescales, differences in diel activities may drive speciation \cite{taylor_role_2017}. However, because nearly all studies to date have been conducted on isolated individuals in laboratory microcosms, the ecological and evolutionary implications of circadian clocks in natural environments remain largely unknown \cite{schwartz_wild_2017}.

While chronobiology requires a physiological and behavioural time scale (\emph{i.e.}, seconds to hours), insect surveys have primarily focused on the phenological scale (\emph{i.e.}, days to months). Compared to bird and mammal studies, where methodological breakthroughs in animal tracking devices have enabled the ecological study of the timing of behaviours, similar tools for invertebrates are lacking\cite{dominoni_methods_2017} or limited to specific cases\cite{brydegaard_daily_2016,goldshtein_long-term_2021,nunes-silva_applications_2019}. Promisingly, portable electronics and machine learning are beginning to reach insect ecology and monitoring\cite{hoye_deep_2021}. For instance, “smart traps” can now automatise traditional insect capture and identification\cite{cardim_ferreira_lima_automatic_2020}. In particular, camera-based traps can passively monitor insects and use deep learning to identify multiple species. However, such tools are often designed for applications on a single focal species and, due to the large amount of data they generate as well as the complexity of the downstream analysis, camera-based traps have typically been limited to daily monitoring and have not previously been used to study insect circadian behaviours.

Here, we present and validate the Sticky Pi, an open-source generalist automatic trap to study insect chronobiology in the field. Our unique framework both automatises insect surveying and adds a novel temporal and behavioural dimension to the study of biodiversity. This work paves the way for insect community chronoecology: the organisation, interaction, and diversity of organisms’ biological rhythms within an ecological community. 


\section*{Results}

\subsection*{Sticky Pi device and Platform}

We built the Sticky Pi (Fig.~\ref{fig:01}A-C), a device that captures insects on a sticky card and images them every twenty minutes. Compared to other methods, our device acquires high-quality images at high frequency, hence providing a fine temporal resolution on insect captures. Devices are equipped with a temperature and humidity sensor and have two weeks of autonomy (without solar panels). Sticky Pis are open-source, 3d printed and inexpensive (<~200 USD). Another unique feature is their camera-triggered backlit flashlight, which enhances the contrast, reduces glare and allows for night-time imaging. As a result, we can discern 3 mm-long insects on a total visible surface of 215 cm² (Fig.~\ref{fig:01}D-E), which is sufficient to identify many taxa. In order to centralise, analyse and visualise the data from multiple devices, we developed a scalable platform (Supplementary Fig.~S\ref{supfig:01}), which includes a suite of services: an Application Programming Interface, a database and an interactive web application (Supplementary Video~1). Deployment and maintenance instructions are detailed in our documentation (\href{https://doc.sticky-pi.com/web-server.html}{https://doc.sticky-pi.com/web-server.html}).

\subsection*{Image processing}
In order to classify captured insects, we developed a novel analysis pipeline, which we validated on a combination of still photographs of standard sticky traps and series of images from ten Sticky Pis deployed in two berry fields for 11 weeks (see Methods section and next result sections). We noticed trapped insects often move, escape, are predated, become transiently occluded, or otherwise decay (Supplementary Video 2). Therefore, we used cross-frame information ‒ rather than independently segmenting and classifying insects frame-by-frame. Our pipeline operates in three steps (summarised below and in Fig.~\ref{fig:02}): 
(i) the “\emph{Universal Insect Detector}” segments insect instances in independent images assuming a two-class problem: insect \emph{vs.} background;
(ii) The “\emph{Siamese Insect Matcher}” tracks insect instances between frames, using visual similarity and displacement;
(iii) The “\emph{Insect Tuboid Classifier}” uses information from multiple frames to make a single taxonomic prediction on each tracked insect instance.

\subsubsection*{Universal Insect Detector}
To segment “insects” from their “background”, we based the Universal Insect Detector on Mask R-CNN \cite{he_mask_2017}, and trained it on 239 hand-annotated images of trapped insects (see Methods section). On the validation dataset, our algorithm had an overall 69.6\% recall and 90.5\% precision (Supplementary Fig.~S\ref{supfig:02}). Noticeably, recall increased to 79.3\% when excluding the 25\% smallest objects (area <~1000 px. \emph{i.e.}, 2.12 mm²), indicating that the smallest insect instances are ambiguous. 

\subsubsection*{Siamese Insect Matcher}
In order to track insects through multiple frames, we built a directed graph for each series of images; connecting instances on the basis of a matching metric, which we computed using a custom Siamese Neural Network (Supplementary Fig.~S\ref{supfig:03}A and Methods section). We used this metric to track insects in a three-pass process (Supplementary Fig.~S\ref{supfig:03}B and Methods section). This step resulted in multi-frame representations of insects through their respective series, which we call “tuboids”. Supplementary Video 3 shows a timelapse video of a series where each insect tuboid is boxed and labelled with a unique number.

\subsubsection*{Insect Tuboid Classifier}

To classify multi-frame insect representations (\emph{i.e.}, “tuboids”), we based the Insect Tuboid Classifier (Fig.~\ref{fig:03}), on a Residual Neural Network (ResNet) architecture\cite{he_deep_2016} with two important modifications: (i) We explicitly included the size of the putative insect as an input variable to the fully connected layer ‒ as size may be important for classification and our images have consistent scale; (ii) Since tuboid frames provide non-redundant information for classification (stuck insects often still move and illumination changes), we applied the convolution layers on six frames sampled in the first 24 h and combined their outputs in a single prediction (Fig.~\ref{fig:03}A). In this study, we trained our classifier on a dataset of 2,896 insect tuboids, trapped in two berry fields in the same location and season (see next result sections and Methods section). We defined 18 taxonomic labels, described in Supplementary Table 1, using a combination of visual identification and DNA barcoding of insects sampled from the traps after they were collected from the field (Supplementary Table 2 and Methods sections). Figure~\ref{fig:03}B-D shows the confusion matrix and summary statistics on the validation dataset (982 tuboids) and representative insect images corresponding to these 18 labels. The overall accuracy (\emph{i.e.}, the proportion of correct predictions) is 78.4\%. Our dataset contained a large proportion of either “Background” objects and “Undefined insects” (16.2\%, 22.4\%, respectively). When merging these two less-informative labels, we reach an overall 83.1\% accuracy on the remaining 17 classes. Precision (\emph{i.e.}, the proportion of correct predictions given a predicted label) and recall (\emph{i.e.}, the proportion of correct prediction given an actual label) were high for the Typhlocybinae (leafhoppers) morphospecies (92\% and 94\%). For \emph{Drosophila suzukii} [Diptera: Drosophilidae] (spotted-wing drosophila), an important berry pest, we labelled males as a separate class due to their distinctive dark spots and also reached a high precision (86\%) and recall (91\%). These results show that performance can be high for small, but abundant and visually distinct taxa.

\subsection*{Sticky Pis can quantify circadian activity in laboratory conditions}
To test whether capture rate on a sticky card could describe the circadian activity of an insect population, we conducted a laboratory experiment on vinegar flies, \emph{Drosophila melanogaster} [Diptera: Drosophilidae], either in constant light (LL) or constant dark (DD), both compared to control populations held in 12:12 h Light:Dark cycles (LD) (Fig.~\ref{fig:04}). From the extensive literature on \emph{D. melanogaster}, we predicted a crepuscular activity LD and DD (flies are “free-running” in DD), but no rhythm in LL\cite{tataroglu_studying_2014}. We placed groups of flies in a large cage that contained a single Sticky Pi (simplified for the laboratory and using infrared light; Methods section). Consistent with previous studies on circadian behaviour of \emph{D. melanogaster}, populations in both LD and DD conditions exhibited strong rhythmic capture rates, with an approximate period of 24 h: 23.8 h and 23.67 h, respectively. For instance, their overall capture rate was approximately 0.6 h$^{-1}$ between ZT22 and ZT23 h, but peaked at 9.5 h$^{-1}$ between ZT01 and ZT02 h (Fig.~\ref{fig:04}D,F). Also as hypothesised, the fly populations held in constant light (LL) showed no detectable behavioural rhythm and had a constant average capture rate of 1.6 h$^{-1}$ (sd = 0.62) (Fig.~\ref{fig:04}C,E). Collectively these observations indicate that Sticky Pis have the potential to capture circadian behaviour in a free-flying insect population.

\subsection*{Sticky Pis quantify activity rhythms of wild \emph{Drosophila suzukii}}

To test the potential of the Sticky Pis to monitor wild populations of free-moving insects in the field, we deployed ten traps in a blackberry field inhabited by the well-studied and important pest species \emph{D. suzukii} (see Methods section). Like \emph{D. melanogaster}, \emph{D. suzukii} has been characterised as crepuscular both in the laboratory\cite{shaw_control_2019} and, with manual observations, in the field\cite{swoboda-bhattarai_diurnal_2020}. Since capture rates can be very low without attractants\cite{swoboda-bhattarai_diurnal_2020}, we baited half (five) of our traps with apple cider vinegar (see Methods section). In addition to \emph{D. suzukii}, we wanted to simultaneously describe the activity of lesser-known species in the same community. In particular, \emph{D. suzukii} and other closely related Drosophila are attacked by parasitoid wasps [Hymenoptera: Figitidae], two of which (\emph{Leptopilina japonica} and \emph{Ganaspis brasiliensis}) have recently arrived in our study region\cite{abram_new_2020}. Their diel activity has not yet been described. In Figure~\ref{fig:05}, we show the capture rate of male \emph{D. suzukii}, other putative Drosophilidae and parasitoid wasps throughout the seven-week trial (Fig.~\ref{fig:05}A) and throughout an average day (Fig.~\ref{fig:05}B). Our results corroborate a distinctive crepuscular activity pattern for male \emph{D. suzukii} and other putative drosophilids. In contrast, Figitidae wasps were exclusively diurnal. Overall, baiting widely increased the number of male \emph{D. suzukii} (from 3.1 to 29.3 $\text{device}^{-1}.\text{week}^{-1}$, $\text{p-value} < 10^{-8}$), and other Drosophilidae (from 11.4 to 54.3 $\text{device}^{-1}.\text{week}^{-1}$, $\text{p-value} < 10^{-9}$), but not parasitoid wasps ($\text{p-value} > 0.9$, Wilcoxon rank-sum tests). These findings indicate that Sticky Pi can quantify the circadian behaviour of a simple insect community in a natural setting.

\subsection*{Sticky Pi, a resource for community chronoecology}

Berry fields are inhabited by a variety of insects for which we aimed to capture proof-of-concept community chronoecological data. In a separate trial, we placed ten Sticky Pis in a raspberry field and monitored the average daily capture rate of eight selected taxa (Fig.~\ref{fig:06}A) over four weeks – we selected these eight taxa based on the number of individuals, performance of the classifier (Fig.~\ref{fig:03}), and taxonomic distinctness (Supplementary Fig.~S\ref{supfig:04} shows the other classified taxa). We then defined a dissimilarity score and applied multidimensional scaling to represent temporal niche proximity in two dimensions (see Methods section). We show that multiple taxa can be monitored simultaneously, and statistically partitioned according to their temporal niche (Fig.~\ref{fig:06}B). Specifically, as shown in Fig.~\ref{fig:06}A, sweat bees (\emph{Lasioglossum laevissimum}), large flies (Calyptratae) and hoverflies (Syrphidae) show a clear diurnal activity pattern with a capture peak at solar noon (Warped Zeitgeber Time = 6h, see Methods section for WZT). Sciaridae gnats were also diurnal, but their capture rate was skewed towards the afternoon, with a peak around WZT = 7h. The Typhlocybinae leafhopper was vespertine, with a single sharp activity peak at sunset (WZT = 11h). The Psychodidae were crepuscular, exhibiting two peaks of activity, at dusk and dawn. Both mosquitoes (Culicidae) and moths (Lepidoptera) were nocturnal. These findings show that, even without \emph{a priori} knowledge of the diel activity of specific taxa, Sticky Pis can inform about both the community structure and temporal patterns of behaviour of a natural insect community.


\section*{Discussion}


We have developed the Sticky Pi, a generalist and versatile insect smart trap that is open-source, documented and affordable (Fig.~\ref{fig:01}). Uniquely, Sticky Pis acquires frequent images to finely describe when specific insects are captured. Since the main limitation to insect chronoecology is the lack of high-frequency population monitoring technologies\cite{dominoni_methods_2017}, our innovation promises to spark discoveries at the frontier between two important domains: chronobiology and biodiversity monitoring. Furthermore, taking multiple images of the same specimen may improve classification performance. To adapt our tool to big-data problems, we designed a suite of web services (Fig.~S\ref{supfig:01}) that supports multiple concurrent users, can communicate with distributed resources and may interoperate with other biodiversity monitoring projects and community science platforms\cite{pocock_chapter_2018}.

Our device’s main limitation is image quality (Fig.~\ref{fig:01}D-E). Indeed, high-performance segmentation and classification of insects were limited to specimens larger than three millimetres (Supplementary Fig.~\ref{fig:02}), hence reducing the taxonomic resolution for small insects. Camera technology is quickly improving and we are currently trialling a version with superior optics (12.3 Mpx, CS mount). We found that segmentation was globally very precise (>90\%) and accurate (80\% for objects larger than 2 mm²). Furthermore, our machine-learning pipeline (Fig.~\ref{fig:02}) showed a high overall accuracy of the Insect Tuboid Classifier (83.1\% on average, when merging background and undefined insects, see Fig.~\ref{fig:03}). 

We corroborated circadian results that had historically been obtained on individually-housed insects, using heterogeneous populations in large flight cages (Fig.~\ref{fig:04}). This suggests that Sticky Pis could be an alternative tool for laboratory experiments on mixed populations of interacting insects. In the field, we monitored both the seasonal and diel activity of a well-studied pest species: spotted-wing drosophila (\emph{D. suzukii}). Like others before\cite{swoboda-bhattarai_diurnal_2020}, we concluded that wild \emph{D. suzukii} was crepuscular (Fig.~\ref{fig:05}). In the process, we also found that Figitidae wasps – natural enemies of \emph{D. suzukii} – were distinctly diurnal and were most often detected later in the season. Finally, we characterised the diel activity of the flying insect community in a raspberry field, without targeting taxa \emph{a priori} (Fig.~\ref{fig:06}). With only ten devices, over 4 weeks, we were able to reveal the diversity of temporal niches; showing coexisting insects with a wide spectrum of diel activity patterns – \emph{e.g.}, diurnal, crepuscular \emph{vs.} nocturnal; bimodal \emph{vs.} unimodal. An additional and noteworthy advantage of time-lapse photography is the incidental description of unexpected behaviours such as insect predation (Supplementary Video~4) and the acquisition of specimens that eventually escape from traps  (Supplementary Video~5).

In the last few years, we have seen applications of chronobiology to fields such as learning\cite{smarr_time_2014} and medicine\cite{cederroth_medicine_2019}. We argue that chronobiological considerations could be equally important to biodiversity conservation and precision agriculture\cite{gottlieb_agro-chronobiology_2019,karapetyan_redox_2018,khyati_insect_2017}. For instance, plants’ defences\cite{goodspeed_arabidopsis_2012,jander_timely_2012} and insecticide efficiency\cite{balmert_time--day_2014,khalid_circadian_2019} may change during the day, implying that agricultural practices could be chronobiologically targeted. In addition, modern agriculture is increasingly relying on fine-scale pest monitoring and the use of naturally occurring biological pest control\cite{gagic_better_2021,tooker_balancing_2020}. Studying insect community chronoecology could help predict the strength of interactions between a pest and its natural enemies, or measure the temporal patterns of recruitment of beneficial natural enemies and pollinators. Monitoring insect behaviours at high temporal resolution is critical for both understanding, forecasting and controlling emerging insect pests in agriculture and, more broadly, to comprehend how anthropogenic activities impact behaviour and biodiversity of insect populations.

\section*{Methods}


\subsection*{Image processing}

\subsubsection*{Universal insect detector}
\paragraph{Data}
We acquired a diverse collection of 309 images by, first, setting standalone sticky cards in the Vancouver area for one to two weeks and taking images with a Sticky Pi afterwards, and, second, by systematically sampling images from the field experiments. The first set of ‘offline’ images was “physically augmented” by taking photographs in variable conditions, which included illumination, presence of water droplets and thin dust particles. We annotated images using Inkscape SVG editor, encoding annotations as SVG paths. The outline of each visible arthropod was drawn. The contours of two adjacent animals were allowed to overlap. We automatically discarded objects smaller than 30 px (\emph{i.e.}, <~2 mm objects that are indiscernible in the images by manual annotators) or wider than 600 px (\emph{i.e.}, objects larger than 40 mm, which were generally artefacts since the vast majority of captured insects are smaller in our study region). Partial insects were only considered if their head and thorax were both visible. This procedure resulted in a total of 16,020 segmented insects.

\paragraph{Training}
To perform instance segmentation, we used Mask R-CNN\cite{he_mask_2017}. In order to train the algorithm, images were pseudo-randomly split into a validation (25\%, 70 images) and a training (75\%, 239 images) set, based on their md5 checksum. Random digital augmentation was applied to all images in the training set: rotation (0, 90, 180, 270 degrees), vertical and horizontal reflections, brightness and contrast (uniform random in [0.9, 1.1]). 

We use the detectron2 implementation \cite{wu_detectron2_2019} of Mask R-CNN to perform the instance segmentation (insect \emph{vs.} background). We retrained a ResNet50 conv4 backbone, with conv5 head, which was pre-trained on the COCO dataset, for 110,000 iterations (14 images per batch) with an initial learning rate of 0.05, decaying by $\gamma=0.8$ every 10,000 iterations.

\paragraph{Generalisation to large images}
The default standard dimension of Mask R-CNN inputs is $1024 \times{} 1024$ px. 
Our images being larger ($2592 \times{} 1944$ px), we performed predictions on 12 $1024 \times{}1024$ tiles (in a $4 \times{} 3$ layout), which mitigates edge effects since tiles overlap sufficiently so that very large insects (>~500 px wide) would be complete in, at least, one tile. A candidate insect instance (defined as a polygon) $B$ was considered valid if and only if $J(A_{i}, B) < 0.5 \forall i$, where $A_i$ represents valid instances in neighbouring tiles, and $J$ is the Jaccard index.

\subsubsection*{Siamese Insect Matcher}
\paragraph{Matching function}
The goal of the Siamese Insect Matcher is to track insect instances through consecutive frames – given that insects may move, escape, be predated, get occluded, etc. The core of the algorithm is the matching function $M(m,n) \in [0, 1]$, between objects $m$ and $n$ detected by the Universal Insect Detector. This section describes how $M(m,n)$ is computed (see also Supplementary Fig.~S\ref{supfig:03}A for a visual explanation).

Given a pair of objects $m, n$, in images $\mathbf{X}^i$ and $\mathbf{X}^j$, we have the binary masks $A_m$ and $A_n$ of $m$ and $n$, respectively.
We then use the same function $D$ to compute two similarity values $S(m,n)$ and $Q(m,n)$. 
With,
$$
S(m,n) = D(\mathbf{X}^i \cap A_m , \mathbf{X}^j \cap A_n )
$$
\emph{i.e.}, the similarity between $m$ in its original frame, $i$, and $n$ in its original frame, $j$. And,
$$
Q(m,n) = D(\mathbf{X}^i \cap  A_m , \mathbf{X}^j \cap A_m )
$$
\emph{i.e.}, the similarity between $m$ in its original frame, $i$, and $m$ projected in the frame $j$. Note, that all inputs are cropped to the bounding box of $A$, and scaled to $105 \times{} 105$ px. $D$ is a Siamese network as defined in\cite{koch_siamese_2015} with the notable distinction that the output of our last convolutional layer has a dimension of $1024 \times{} 1$ (\emph{vs.} $4096 \times{} 1$, in the original work), for performance reasons. The two resulting similarity values, $S(m,n)$ and $Q(m,n)$ are then processed by a small, custom, four-layers, fully connected neural network, $H(I)$. The inputs are: 
$$
I=\{ S(m,n), Q(m,n), d(C(m), C(n)), |log(\mathcal{A}_m/\mathcal{A}_n)|, log(\Delta t + 1)\}
$$ Where $d$ is the Euclidean distance between the centroids $C$. $\mathcal{A}$ is the area of an object, and $\Delta t = t_j- t_i$. Our four layers have dimensions {5,4,3,1}. We use a ReLU activation function after the first two layers and a sigmoid function at the output layer.

\paragraph{Data}
In order to train the Siamese Insect Matcher core Matching function $M$, we first segmented image series from both berry field trials (see below) with the Universal Insect Detector to generate annotations (see above). We randomly sampled pairs of images from the same device, with the second image between 15 min and 12 h after the first one. We created a composite SVG image that contained a stack of the two images, and all annotations as paths. We then manually grouped (\emph{i.e.}, SVG groups) insects that were judged the same between the two frames. We generated 253 images this way, containing a total of 7238 positive matches. Negative matches (N = 154,842) were defined as all possible non-positive matches between the first and second images. Since the number of negatives was very large compared to the positive matches, we biased the proportion of negative matches to 0.5 by random sampling during training.

\paragraph{Training}
We trained the Siamese Insect Matcher in three steps. First, we pre-trained the Siamese similarity function $D$ by only considering the $S(m,n)$ branch of the network (\emph{i.e.}, apply the loss function on this value). Then we used the full network, but only updated the weights of the custom fully connected part $H(I)$. Last, we fine-tuned by training the entire network. For these three steps, we used Adaptive Moment Estimation with learning rates of $2\times{}10^{-5}$, $0.05$, and $2\times{}10^{-5}$, for 500, 300, and 500 rounds, respectively. We used a learning rate decay of $\gamma = 1- 10^{-3}$ between each round. Each round consisted of a batch of 32 pairs. We defined our loss function as binary cross-entropy.

\paragraph{Tracking}
We then use our instance overall matching function ($M$) for tracking insects in three consecutive steps. We formulate this problem as the construction of a graph $G(V, E)$, with the insect instances in a given frame as vertices $V$, and connection to the same insect, in other frames, as edges $E$ (see also Supplementary Fig.~S\ref{supfig:03}B for a visual explanation). This graph is directed (through time), and each resulting (weakly) connected subgraph is an insect “tuboid” (\emph{i.e.}, insect instance). Importantly, each vertex can only have a maximum of one incoming and one outgoing edge. That is, given $v \in V$, $\text{deg}^-(v) \le 1$ and $\text{deg}^+(v) \le 1$. We build $G$ in three consecutive steps. 

First, we consider all possible pairs of instances $m, n$ in pairs of frames $i, j$, with $j=i+1$ and compute $M(m,n)$. In other words, we match only in contiguous frames. Then, we define a unique edge from vertex $m$ as:

\begin{equation}
	\label{eq:01}
	e = \begin{cases}
	\emptyset~\text{if}~M(m,n)<k \forall n \\
	\{(m, \underset{n}{\operatorname{arg\,max}}\,M(m,n))\}~\text{else} \\
	\end{cases}
\end{equation}

Where $k=0.5$ is a threshold on $M$. That is, we connect an instance to the highest match in the next frame, as long as the score is above 0.5. We obtain a draft network with candidate tuboids as disconnected subgraphs.


Second, we apply the same threshold (eq.~\ref{eq:01}), and consider the pairs all pairs $m, n$, in frames $i,j$, where $\text{deg}^+(m) = 0$, $\text{deg}^-(n) = 0$, $j - i > 1$ and $t_j - t_i < 12h$. That is, we attempt to match the last frame of each tuboid to the first frame of tuboids starting after. We perform this operation recursively, always connecting the vertices with the highest overall matching score, and restarting. We stop when no more vertices match. This process bridges tuboids when insects were temporarily undetected (\emph{e.g.}, occluded).

Finally, we define any two tuboids $P(E, V)$ and $Q(F, W)$ (\emph{i.e.}, disconnected subgraphs, with vertices $V$ and $W$, and edges $E$ and $F$) as “conjoint” if and only if $t_{v} \neq t_{w} \forall v,w$, and $min(t_v) \in [min(t_w), max(t_w)]$ or $min(t_w) \in [min(t_v), max(t_v)]$. That is, two tuboids are conjoint if and only if they overlap in time, but have no coincidental frames. We compute an average score between conjoint tuboids as:
$$
\bar{M}(P, Q) = \frac{1}{N}\sum_{v,w \in K}^{}M(v,w)
$$
Where $K$ is the set of $N$ neighbouring pairs, in time: 
$$
K = \bigcup_{v} \{(v,  \underset{w \forall t_v > t_w}{\operatorname{arg\,min}}(t_v - t_w)), (v,\underset{w \forall t_v < t_w}{\operatorname{arg\,min}}(t_w - t_v)\}
$$
That is, the average matching score between all vertices and their immediately preceding and succeeding vertices in the other tuboid. We apply this procedure iteratively with a threshold $k=0.25$, merging first the highest-scoring pair of tuboids. Finally, we eliminate disconnected subgraphs that do not have, at least, four vertices.

\subsubsection*{Insect Tuboid Classifier}
\paragraph{Data}
We generated tuboids for both field trials (see below) using the Siamese Insect Matcher described above. We then visually identified and annotated a random sample of 4003 tuboids. Each tuboid was allocated a composite taxonomic label as “type/order/family/genus/species”. 
Type was either Background (not a complete insect), Insecta or Ambiguous (segmentation or tracking error). It was not possible to identify insects at a consistent taxonomic depth. Therefore, we characterised tuboids at a variable depth (\emph{e.g.}, some tuboids are only “Insecta/*” while others are “Insecta/Diptera/Drosophilidae/Drosophila/D. suzukii”).

\paragraph{Training}
In order to train the Insect Tuboid Classifier, we defined 18 “flat” classes (\emph{i.e.}, treated as discrete levels rather than hierarchical, see Fig.~\ref{fig:03}). We then pseudo-randomly (based on the image md5 sum) allocated each tuboid to either the training or the validation data subset, ensuring an approximate ¾ to ¼, training to validation, ratio, per class. We excluded the 125 ambiguous annotations present, resulting in a total of 2896 training and 982 validation tuboids. 
We initialised the weight of our network from a ResNet50 backbone, which had been pre-trained on a subset of the COCO dataset. For our loss function, we used cross-entropy, and stochastic gradient descent as an optimizer. We set an initial learning rate of 0.002 with a decay $\gamma = 1- 10^{-4}$ between each round and a momentum of 0.9. A round was a batch of eight tuboids. Each individual image was augmented during training by adding random brightness, contrast and saturation, randomly flipping along x and y axes and random rotation [0, 360]°. All individual images were scaled to $224 \times{} 224$ px. Batches of images we normalised during training (standard for ResNet). We trained our network for a total of 50,000 iterations.

\subsubsection*{Implementation, data and code availability}
We packaged the source code of the image processing as a python library: sticky-pi-ml (https://github.com/sticky-pi/sticky-pi-ml). Our work makes extensive use of scientific computing libraries OpenCV (Bradski, 2000), Numpy\cite{harris_array_2020}, PyTorch\cite{paszke_pytorch_2019}, sklearn\cite{pedregosa_scikit-learn_2011}, pandas\cite{the_pandas_development_team_pandas-devpandas_2020} and networkx\cite{hagberg_exploring_2008}. Neural network training was performed on the Compute Canada platform, using a NVidia 32G V100 GPU. The dataset, configuration files and resulting models for the Universal Insect Detector, the Siamese Insect Matcher and the Insect Tuboid Classifier are publicly available, under the creative commons license\cite{geissmann_sticky_2021}. 

\subsection*{Laboratory experiments}
In order to reproduce classic circadian experiments in an established model organism, we placed approximately 1500 CO2-anesthetized Drosophila melanogaster individuals in a 950 mL (16 oz) deli container, with 100 mL of agar (2\%), sucrose (5\%) and propionic acid (0.5\%) medium. The top of this primary container was closed with a mosquito net, and a 3 mm hole was pierced on its side, 40mm from the bottom, and initially blocked with a removable cap. Each cup was then placed in a large (25×25×50 cm) rectangular cage (secondary container), and all cages were held inside a temperature-controlled incubator. In the back of each cage, we placed a Sticky Pi device that had been modified to use infrared, instead of visible, light. In addition, we placed 100 mL of media in an open container inside each cage, so that escaping animals could freely feed. Flies were left at least 48h to entrain the light regime and recover from anaesthesia before the small aperture in their primary container was opened. The small diameter of the aperture meant that the escape rate, over a few days, was near-stationary.
The D. melanogaster population was a mixture of CantonS males and females from three to five days old, and the number of individuals was approximated by weighting animals (average fly weight = $8.4×10^{-4}$g). During the experiments, the temperature of the incubators was maintained at 25°C and the relative humidity between 40 and 60\%. All animals were entrained in a 12:12 h Light:Dark regime. Flies were kindly given by Mike Gordon (University of British Columbia). One experimental replicate (device × week) was lost due to a sticky card malfunction.

\subsection*{Field experiments}
In order to test the ability of the Sticky Pi device to capture the daily activity patterns of multiple species of free-living insects, we deployed ten prototype devices on an experimental farm site in Agassiz, British Columbia, Canada (GPS: 49.2442, -121.7583) from June 24 to September 30, 2020. The experiments were done in two plots of berry plants, raspberries and blackberries, which mature and decline during early and late summer, respectively. Neither plot was sprayed with pesticides at any point during the experiments.

\subsubsection*{Blackberry field}
The blackberry (\emph{Rubus fruticosis} var. ‘Triple Crown’) plot was made up of five rows, each of which was approximately 60 metres long. Each row had wooden posts (approximately 1.6 m high) spaced approximately 8.5 m apart, along which two metal ‘fruiting wires’ were run at two different heights (upper wire: 1.4 m; lower wire: 0.4 m). Two traps, one baited with apple cider vinegar and one unbaited, were set up on two randomly selected wooden posts within each of the five rows, with the position of baited and unbaited traps (relative to the orientation of the field) alternated among rows. A plastic cylindrical container (diameter: 10.6 cm; height: 13.4 cm) with two holes cut in the side (approximately $3 \times{} 6$ cm) and fine mesh (knee-high nylon pantyhose) stretched over the top, containing approximately 200 mL of store-bought apple cider vinegar was hung directly under baited traps
(Supplementary Fig.~S\ref{supfig:05}). No such container was hung under unbaited traps. Vinegar in the containers hung under baited traps was replaced weekly. Traps were affixed to the wooden posts at the height of the upper fruiting wire so that they faced southwards. Trapping locations did not change over the course of the experiment, which began approximately two weeks after the beginning of blackberry fruit ripening (August 12, 2020) and ended when fruit development had mostly concluded (September 30, 2020). Sticky cards were replaced once weekly, and photographs were offloaded from traps every 1–2 weeks. Approximately 15 trap-days of data were lost during the experiment due to battery malfunctions. Overall, 475 trap-days over 70 replicates (device $\times{}$ week), remained (\emph{i.e.}, 96.9\%).

\subsubsection*{Raspberry field}
Ten Sticky Pi devices were set up in a raspberry (\emph{Rubus idaeus} var. ‘Rudi’) plot with 6 rows, each of which was approximately 50 m long. Each row had wooden posts (approximately 1.6 m high) spaced 10 m apart, along which two metal ‘fruiting wires’ were run at two different heights (upper wire: 1.4 m; lower wire: 0.4 m) to support plants. Two traps were set up on a randomly selected wooden post within each of 5 randomly selected rows. At each location, to capture any potential fine-scale spatial variation in insect communities, traps were affixed to the wooden posts at two different heights; at the same levels as the upper and lower fruiting wires. Traps were oriented southwards. Trapping locations within the field did not change over the course of the experiment, which began approximately one week after the beginning of raspberry fruit ripening (June 24, 2020) and ended after fruiting had concluded (July 29, 2020). Sticky cards were replaced once weekly, and photographs were offloaded from traps every 1-2 weeks. Some data (approximately nine days, from three replicates) were lost due to battery malfunctions. Overall, 271 trap-days over 40 replicates (device $\times$ week), remained (\emph{i.e.}, 96.8\%).

\subsubsection*{DNA barcoding}
In order to confirm the taxonomy of visually identified insects, we recovered specimens from the sticky cards after the trials to analyse their DNA and inform visual labelling. We targeted the molecular sequence of the cytochrome c oxidase subunit I (CO1). The overall DNA barcoding workflow follows established protocols\cite{dewaard_expedited_2018} with minor modifications. Briefly, the genomic DNA of insect specimens was extracted with the QIAamp Fast DNA Stool Mini Kit (QIAGEN) according to the manufacturer’s instructions. The resulting DNA samples were then subjected to concentration measurement by a NanoDrop™ One/OneC Microvolume UV-Vis Spectrophotometer (Thermo Fisher Scientific) and then normalised to a final concentration of 50 ng/µl. Next, depending on the identity of the specimen, the following primer pairs were selected for CO1 amplification: C\_LepFolF/C\_LepFolR\cite{hernandeztriana_recovery_2014}, MHemF/LepR1\cite{park_barcoding_2011}. Amplification of the CO1 barcode region was conducted using Phusion® High-Fidelity DNA Polymerase (New England Biolabs Inc.) with the following 25 µl reaction recipe: 16.55 µl ddH2O, 5 µl 5HF PCR buffer, 2 µl 2.5 mM dNTP, 0.6 µl of each primer (20 µM), 0.25 µl Phusion polymerase and finally 2 µl DNA template. All PCR programs were set up as the following: 95°C for 2 min; 5 cycles at 95°C for 40 s, 4°C for 40 s, and 72°C for 1 min; then 44 cycles at 95°C for 40 s, 51°C for 40 s, and 72°C for 1 min; and a final extension step at 72 °C for 10 min. PCR products were then subjected to gel electrophoresis and then purified with EZ-10 Spin Column DNA Gel Extraction Kit (Bio Basic). After Sanger sequencing, a Phred score cutoff of 20 was applied to filter out poor-quality sequencing reads. 
The barcode index number (BIN) of each specimen was determined based on 2\% or greater sequence divergence applying the species identification algorithm available on the Barcode of Life Data Systems (BOLD) version 4 (Ratnasingham and Hebert, 2013). Barcode sequences will be deposited in GenBank (Accession nos.~XXXX-XXXX). We also took several high-quality images of each specimen before DNA extraction and embedded them in a single table (Supplementary Table 2) to cross-reference morphology and DNA sequences (Bianchi and Gonçalves, 2021).


\subsection*{Statistics and data analysis}

\subsubsection*{Laboratory}
Images from the laboratory trials were processed using a preliminary version of the Universal Insect Detector on independent frames – \emph{i.e.}, without subsequent steps. This resulted in a raw number of detected insects on each independent frame. In order to filter out high-frequency noise in the number of insects, we applied a running median filter (k = 5) on the raw data. Furthermore, we filtered the instantaneous capture rate ($dN/dt$) with a uniform linear filter (k = 5). These two operations act as a low-pass frequency filter, with an approximate span of 100 min.

\subsubsection*{Warped Zeitgeber Time}
Zeitgeber Time (ZT) is conventionally used to express the time as given by the environment (typically light, but also temperature, etc). By convention, ZT is expressed in hours, between 0 and 24, as the duration since the onset of the day (\emph{i.e.}, sunrise = ZT0). Using ZT is very convenient when reporting and comparing experiments in controlled conditions. However, ZT only defines a lower bound (ZT0) and is therefore difficult to apply when day length differs (which is typical over multiple days, under natural conditions, especially at high and low latitudes). In order to express time relatively to both sunrise and sunset, we applied a simple linear transformation of ZT to Warped Zeitgeber Time (WZT), $W(z)$.

Like ZT, we set WZT to be 0 at sunrise, but to always be ½ day at sunset, and to scale linearly in between. Incidentally, WZT is ¼ day and ¾ day at solar noon and at solar midnight, respectively.
Formally, we express WZT as a function of ZT with:

$$W(z)= \begin{cases}
	az  &,  \text{if } z \leq d  \\
	a'z + b' &,  \text{otherwise}
\end{cases}$$
Where, $a$, $a'$ and $b'$ are constants, $d$ is the day length, as a day fraction. $z \in [0,1)$ is ZT and can be computed with $z = t - s \mod 1$, where $t$ is the absolute time and $s$, the time of the sunrise. Since WZT is 1 when ZT is 1, we have:
$W(1) = a' 1 + b' = 1$
Also, WZT is ½ at sunset: $W(d) = ad = a’d +b’ = \frac{1}{2}$, 
Therefore, $a = \frac{1}{2 d}$, $a' = \frac{1}{2 (1 - d)}$ and $b' = 1 - a'$.
	
\subsubsection*{Multidimensional scaling}

We derived the distance $d$ between two populations, $x$ and $y$ from the Pearson correlation coefficient $r$, as $d_{xy} = \frac{1 - r_{xy}}{2}$. In order to assess the sensitivity of our analysis, we computed 500 bootstrap replicates of the original data by random resampling of the capture instances, with replacement, independently for each taxon. We computed one distance matrix and multidimensional scaling (MDS) for each bootstrap replicate and combined the MDS solutions (Jacoby and Armstrong, 2014). The 95\% confidence ellipses were drawn assuming a bivariate t-distribution.

\subsubsection*{Implementation and code availability}

Statistical analysis and visualisation were performed in R 4.0\cite{r_core_team_r_2021}, with the primary use of packages, smacof\cite{leeuw_multidimensional_2009}, data.table\cite{dowle_datatable_2020}, mgcv\cite{wood_generalized_2017}, maptools\cite{bivand_maptools_2020}, ggplot2\cite{wickham_ggplot2_2016}, 
rethomics\cite{geissmann_rethomics_2019}. 
The source code to generate the figures is available at \href{https://github.com/sticky-pi/sticky-pi-manuscript}{https://github.com/sticky-pi/sticky-pi-manuscript}.

{\setstretch{1.0}
\bibliography{main.bib}
}

%\noindent LaTeX formats citations and references automatically using the bibliography records in your .bib file, which you can edit via the project menu. Use the cite command for an inline citation, e.g.  \cite{Hao:gidmaps:2014}.

%For data citations of datasets uploaded to e.g. \emph{figshare}, please use the \verb|howpublished| option in the bib entry to specify the platform and the link, as in the \verb|Hao:gidmaps:2014| example in the sample bibliography file.



\section*{Acknowledgements}
{\setstretch{1.0}
	
We thank all members of the Plant-Insect Ecology and Evolution Laboratory (UBC) and the Insect Biocontrol laboratory (AAFC) for their help. In particular, Warren Wong (UBC/AAFC), Matt Tsuruda (UBC), Dr. Pierre Girod (UBC), Sara Ng (UBC), Yuma Baker (UBC), Jade Sherwood (AAFC/University of the Fraser Valley), Jenny Zhang (UBC/AAFC) for helping with design decisions, tedious image annotations, the literature search and the design of the lab experiments. We thank Dr. Mike Gordon (UBC) for providing us with Drosophila melanogaster CS flies. We are very grateful to Dr. Mark Johnson (UBC), Dr. Esteban Beckwith (Imperial College London), Dr. Alice French (Imperial College London), Luis García Rodríguez (Universität Münster), Mary Westwood (University of Edinburgh) and Dr. Lucia Prieto (Francis Crick Institute) for their very constructive advice and discussions on various aspects of the project. We thank Devika Vishwanath, Samia Siddique Sama and Priyansh Malik, students of the Engineering Physics program (UBC) as well as their mentors for their ongoing work on the next version of the Sticky Pi. The Compute Canada team provided remarkable support and tools for this project. Q.G. was funded by the International Human Frontier Science Program Organization (LT000325/2019). This research (funding to P.K.A., C.H.H. and J.C.) is part of Organic Science Cluster 3, led by the Organic Federation of Canada in collaboration with the Organic Agriculture Centre of Canada at Dalhousie University, supported by Agriculture and Agri-Food Canada’s Canadian Agricultural Partnership - AgriScience Program. PKA was supported by funding from Agriculture and Agri-Food Canada. This work was also supported by a Seeding Food Innovation grant from George Weston Ltd. to CHH and JC, and a Canada Research Chair award to CHH. We acknowledge that some of the research described herein occurred on the traditional, ancestral, and unceded \musqueam Musqueam territory and on the traditional lands of the Sto:lo people.}

\section*{Author contributions statement}
Q.G., P.K.A. and J.C. conceived the experiments.
Q.G. oversaw the image dataset annotations.
Q.G. conducted the laboratory experiment. 
P.K.A. conducted the field experiments.
D.W. characterised the insect vouchers (DNA and images).
Q.G. analysed the results.
All authors wrote and reviewed the manuscript. 

\section*{Additional information}
The authors declare no competing interests


\pagebreak
\begin{figure}[ht]
\centering
\includegraphics[width=0.75\linewidth]{fig/01_hardware/hardware.png}
\caption{\textbf{Sticky Pi device.} \textbf{A-B}, Assembled Sticky Pi. The device dimensions are $326 \times{} 203 \times{} 182$ mm ($d \times{} w \times{} h$). \textbf{C}, Exploded view, showing the main hardware components. Devices are open-source, affordable and can be built with off-the-shelf electronics and a 3d printer. Each Sticky Pi takes an image every 20 minutes using an LED-backlit flash. \textbf{D}, Full-scale image as acquired by a Sticky Pi (originally $1944 \times{} 2592$ px, $126 \times{} 168$ mm ). E, Magnification of the $500 \times{} 500$ px region shown in D.}
\label{fig:01}
\end{figure}
\pagebreak

\begin{figure}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/02_processing_workflow/processing_workflow.png}
	\caption{\textbf{Overview of the Image processing workflow.} Devices acquire images approximately every 20 minutes, which results in a 500 image-long series per week per device. Rows in the figure represent consecutive images in a series. Series are analysed in three main algorithms (left to right). Firstly, the \emph{Universal Insect Detector} applies a two-class Mask R-CNN to segment insect instances (\emph{vs.} background), blue. Secondly, the \emph{Siamese Insect Matcher} applies a custom Siamese-network-based algorithm to track instances throughout the series (red arrows), which results in multiple frames for the same insect instance, \emph{i.e.}, “insect tuboids”. Lastly, the \emph{Insect Tuboid Classifier} uses an enhanced ResNet50 architecture to predict insect taxonomy from multiple photographs.}
	\label{fig:02}
\end{figure}

\pagebreak

\begin{figure}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/03_itc/itc.png}
	\caption{\textbf{Insect Tuboid Classifier description and performance}. \textbf{A}, Algorithm to classify insect tuboids. The first image as well as five randomly selected within the first day of data are selected. Each image is scaled and processed by a ResNet50 network to generate an output feature vector per frame. Each vector is augmented with the original scale of the object, and the element-wise median over the six frames is computed. The resulting average feature vector is processed by a last, fully-connected, layer with an output of 18 labels. \textbf{B}, Confusion matrix on the validation dataset for the 18 labels, and a total of 982 tuboids. The rows show the class labels as annotated in the dataset, and the columns show the labels predicted by the trained algorithm. \textbf{C}, Summary of B, showing precision, recall and f1-score for each label. \textbf{D}, Representative examples of the 18 different classes, numbers match rows in B and C. Abbreviated rows in B and C are \emph{Macropsis fuscula} (3), \emph{Drosophila suzukii} males (4), drosophilids that are not male \emph{D. suzukii} (5), \emph{Anthonomus rubi} (11), \emph{Psyllobora vigintimaculata} (12), Coleoptera that do not belong to any above subgroup (14) and \emph{Lasioglossum laevissimum} (16).}
	\label{fig:03}
\end{figure}

\pagebreak

\begin{figure}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/04_dmel_lab/dmel_lab.png}
	\caption{\textbf{Sticky Pis can monitor circadian rhythms of free-moving populations in the laboratory}. Vinegar flies, \textbf{Drosophila melanogaster}, were held in a large cage with a Sticky Pi. We conducted two experiments to show the effect of Light:Light (red; A, C, E) and Dark:Dark (blue; B, D, F) light-regimes on capture rate. Each was compared to a control population that remained in the entrainment conditions: Light:Dark, 12:12 h cycles (black). \textbf{A-B}, Cumulative number of insects captured over time. Columns of the panels correspond to independent full replicates. We used two devices per condition, in each full replicate. \textbf{C-D}, Capture rates over circadian time. As expected, capture rates in LD and DD show a clear crepuscular activity, but no activity peak in constant light. \textbf{E-F}, Autocorrelation of capture rates. Each thin line represents a series (\emph{i.e.}, one device in one full replicate), and the thick line is the average autocorrelogram. The green dotted line shows the expectation under the hypothesis that there is no periodic pattern in capture rate (ACF: AutoCorrelation Function).
	}
	\label{fig:04}
\end{figure}

\pagebreak

\begin{figure}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/05_dsuz_field/dsuz_field.png}
	\caption{\textbf{Sticky Pis can monitor spotted-wing drosophila diel activity in the field.} We deployed ten Sticky Pis in a blackberry field for seven weeks and attached apple-cider-vinegar baits to half of them (blue \emph{vs.} red for unbaited control). This figure shows specifically the males \textbf{Drosophila suzukii}, the other Drosophilidae flies and the Figitidae wasps. \textbf{A}, Capture rate over time, averaged per day, showing the seasonal occurrence of insect populations. \textbf{B}, Average capture rate over time of the day (note that time was transformed to compensate for changes in day length and onset – \emph{i.e.}, Warped Zeitgeber Time: 0h and 12h represent the sunset and sunrise, respectively, see Material and Methods). Both males \emph{D. suzukii} and the other drosophilids were trapped predominantly on the baited devices. Both populations exhibit a crepuscular activity. In contrast, Figitidae wasps have a diurnal activity pattern and are unaffected by the bait. Error bars show standard errors across replicates ($\text{device} \times{} \text{week}$).}
	\label{fig:05}
\end{figure}

\pagebreak

\begin{figure}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/06_temporal_niches/temporal_niches.png}
	\caption{\textbf{Sticky Pi reveals community chronoecology.} In order to assess the activity pattern of a diverse community, we deployed ten Sticky Pis in a raspberry field for four weeks, replacing sticky cards weekly. This figure shows a subset of abundant insect taxa that were detected by our algorithm with high precision (see supplementary material for the full dataset). \textbf{A}, Average capture rate over time of the day (note that time was transformed to compensate for changes in day length and onset – \emph{i.e.}, Warped Zeitgeber Time: 0 h and 12 h represent the sunset and sunrise, respectively. See Material and Methods). \textbf{B}, Multidimensional scaling of the populations shown in A. Similarity is based on the Pearson correlation between the average hourly activity of any two populations. Small points are individual bootstrap replicates, and ellipses are 95\% confidence intervals (see Material and Methods). Insect taxa partition according to their temporal activity pattern (\emph{e.g.}, nocturnal, diurnal or crepuscular). Error bars in A show standard errors across replicates ($\text{device} \times{} \text{week}$). Individual facets in A were manually laid out to match the topology of B.}
	\label{fig:06}
\end{figure}
\pagebreak


\begin{supfig}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/S_fig_01_platform/platform.png}
	\caption{\textbf{The Sticky Pi platform.} Sticky Pi devices acquire images that are retrieved using a “data harvester” – based on another Raspberry Pi. The data from the harvesters are then incrementally uploaded to a centralised, per-laboratory, database. Images are then automatically pre-processed (the Universal Insect Detector is applied). Users and maintainers can visualise data in real-time using our Rshiny web application. The remote Application Programming Interface (API) is secured behind an Nginx server, and the images are saved on an S3 server. All components of the server are deployed as individual interacting Docker containers. API documentation and source code are available on \href{https://doc.sticky-pi.com/web-server.html}{https://doc.sticky-pi.com/web-server.html}.}
	\label{supfig:01}
\end{supfig}
\pagebreak


\begin{supfig}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/S_fig_02_uid/uid.png}
	\caption{\textbf{Validation of the Universal Insect Detector.} The Universal Insect Detector performs instance segmentation, using Mask R-CNN, on all images in order to detect insects \emph{vs.} background. \textbf{A}, representative image region from the validation dataset. \textbf{B}, Manual annotation of A. \textbf{C}, Automatic segmentation of A. Coloured arrows show qualitative differences between human label (B) and our algorithm: either false positives or false negatives, in blue and red, respectively. Note that often, samples are degraded and ambiguous, even for trained annotators. \textbf{D}, Recall as a function of insect size, showing our method is more sensitive to larger insects. \textbf{E}, Precision as a function of insect size. Precision is overall stationary. The panels on top of D and E show the marginal distribution of insect areas as a histogram. Both curves on D and E are Generalised Additive Models fitted with binomial response variables. Ribbons are the models’ standard errors. The validation dataset contains a total of 5965 insects.}
	\label{supfig:02}
\end{supfig}
\pagebreak

\begin{supfig}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/S_fig_03_sim/sim.png}
	\caption{\textbf{Description of the Siamese Insect Matcher.} \textbf{A}, The matching metric in the Siamese Insect Matcher is based on a Siamese network (see Material and Methods). \textbf{B}. The resulting score, $M$, is used in three steps to connect insect instances between frames. The algorithm results in a series of tuboids, which are representations of single insects through time.}
	\label{supfig:03}
\end{supfig}
\pagebreak

\begin{supfig}[ht]
	\centering
	\includegraphics[width=1.0\linewidth]{fig/S_fig_05_temporal_niche_complete/temporal_niches_complete.png}
	\caption{\textbf{Temporal niches of insect taxa in a raspberry field community.} Complementary data to Fig. 6, showing all predicted taxa. Full species names are in the legend of Fig. 3 and in the result section. The low relative frequency of \emph{Drosophila suzukii} in this unbaited trial and visual inspection suggest male \emph{D. suzukii} are false positives. Other drosophilid-like flies appear to be unknown small diurnal Diptera.}
	\label{supfig:04}
\end{supfig}

\pagebreak

\begin{supfig}[ht]
	\centering
	\includegraphics[width=0.9\linewidth]{fig/S_fig_06_baiting_device/bait_with_trap.png}
	\caption{\textbf{Baited sticky pi.} Sticky pi device (top) with an olfactory bait (bottom). The bait consists of a container holding 200mL of apple cider vinegar protected behind a thin mesh. Apple cider vinegar was replaced weekly during trap maintenance.}
	\label{supfig:05}
\end{supfig}


\clearpage
\newpage
\section*{Supplementary material}

\paragraph*{Supplementary Table 1}
\textbf{Description of the 18 taxonomical labels.} We selected 18 taxa as discrete labels based on both visual examination and DNA barcoding evidence.This table describes the selected groups.
\href{https://figshare.com/s/999a3b7a3c23a050b106}{Data on figshare.}

\paragraph*{Supplementary Table 2}
\textbf{Representative insect vouchers used for DNA barcoding.} 
Data is compiled as an excel spreadsheet with embedded images of the specimens of interest alongside the DNA sequence of their cytochrome c oxidase subunit I (CO1). their DNA-inferred taxonomy and, when relevant, some additional notes. 
The printed labels in the images are $5 \times{} 1$ mm wide. The column “label\_in\_article” corresponds to the visually distinct group to which insects were allocated.
\href{https://figshare.com/s/bdb7bef8c26065e3dd17}{Data on figshare.}

\paragraph*{Supplementary Video 1}
\textbf{The Sticky Pi web application.} Web interface of the Sticky Pi cloud platform. Users can login and select a data range and devices of interest. Then an interactive plot shows environmental conditions over time for each device. Hovering on graphs shows a preview of the images at that time. Users can click on a specific time point to create a pop-up slideshow with details about the image as well as preprocessing results (number and position insects).
\href{https://figshare.com/s/e901a8943d4459cb8f08}{Data on figshare.}


%Supplementary Video S2 Preview at https://youtu.be/wn5Ii7RCrgE 
\paragraph*{Supplementary Video 2}
\textbf{Typical image series.} Video showing one week of data at 10 frames per second. Images are sampled approximately every 20 minutes. Note the variation of lighting, transient occlusions and insects escaping or degrading.
\href{https://figshare.com/s/02944865bfd645047355}{Data on figshare.}

\paragraph*{Supplementary Video 3}
% Supplementary Video S3 Preview at https://www.youtube.com/watch?v=pImE1W48I-E
\textbf{Output of the Siamese Insect Matcher.} Each rectangle is a bounding box of an inferred insect instance.
\href{https://figshare.com/s/096d4a80b93f8380c156}{Data on figshare.}

%Supplementary Video S4 Preview at https://youtu.be/h4CMcmLKwNU
\paragraph*{Supplementary Video 4}
\textbf{Predation of trapped insects by gastropods.} Video showing the extent of slug predation on trapped insects in our dataset. Each rectangle is a bounding box of an inferred insect instance (\emph{i.e.}, tuboid).
\href{https://figshare.com/s/889c8af79a7f90db263e}{Data on figshare.}



%Supplementary Video S5:  Preview at https://youtu.be/efpmVU9kKck 
\paragraph*{Supplementary Video 5}
\textbf{\emph{Anthonomus rubi} escaping a sticky trap.} Multiple individual strawberry blossom weevils (\emph{A. rubi}) impact the trap, but the majority rapidly manage to escape. \emph{A. rubi} is an emergent invasive pest in North America.
\href{https://figshare.com/s/0872dcd897ef7b82d1a3}{Data on figshare.}


\end{document}